test/qpy_compat/process_version.sh

```diff
@@ -43,21 +43,27 @@ fi
 
 if [[ ! -d qpy_$version ]] ; then
     echo "Building venv for qiskit-terra $version"
-    python -m venv $version
+    python -m venv "$version"
     if [[ ${parts[0]} -eq 0 ]] ; then
-        ./$version/bin/pip install -c qpy_test_constraints.txt "qiskit-terra==$version"
+        if ! "./$version/bin/pip" install -c qpy_test_constraints.txt "qiskit-terra==$version"; then
```

Comment:
I worry about this masking real issues because of a spurious network failure or something. This will make a failure to install a historical version a skip instead of a failure which has the risk of glossing over real issues. Especially if we were to introduce some incompatibility with that version. I'm not sure if there is a better way to detect a missing version on pypi though, the only thing I can think of is maybe querying the pypi api at: https://pypi.org/simple/qiskit and https://pypi.org/simple/qiskit-terra/

Reply:
Most of the time we don't actually entire this block at all, because we cache the historical environments, so this should largely only trigger when we're in the intermediate period when a version is partially uploaded, and the environments haven't been created yet.

I'm fine if we want to have some other detection that a release we uploaded to PyPI fails to install, but I don't think it should be part of the QPY tests, with them being in the all-PRs merge pipeline.

Reply:
I've seen a fair number of network issues causing QPY test failures, the caching doesn't seem to work very reliably. My concern here is in the presence of one of those network failures instead of failing we'd skip that version and then lose the coverage the testing is supposed to provide. That opens up an opportunity for a regression/backwards incompatibility to slip in.

Reply:
That's fair enough.  I'm ok pivoting this PR to make a more informed decision about which sets of versions it's even going to attempt to install - we already do a bit of that with the pre-releases, and it's easy enough as you say to turn that logic into a slightly fancier script that asks PyPI for the list of versions (and limit it to those that have the Linux wheel0, rather than us trying to infer it from the repository tags.

Reply:
Ok, this should now hopefully be done correctly in 394b6c1.